vector-stores/spring-ai-pgvector-store/src/main/java/org/springframework/ai/vectorstore/PgVectorStore.java

```diff
@@ -535,12 +558,35 @@ public Builder withInitializeSchema(boolean initializeSchema) {
 			return this;
 		}
 
+		public Builder withObservationRegistry(ObservationRegistry observationRegistry) {
+			this.observationRegistry = observationRegistry;
+			return this;
+		}
+
+		public Builder withSearchObservationConvention(
+				VectorStoreObservationConvention customSearchObservationConvention) {
+			this.searchObservationConvention = customSearchObservationConvention;
+			return this;
+		}
+
 		public PgVectorStore build() {
 			return new PgVectorStore(schemaName, vectorTableName, vectorTableValidationsEnabled, jdbcTemplate,
 					embeddingModel, dimensions, distanceType, removeExistingVectorStoreTable, indexType,
-					initializeSchema);
+					initializeSchema, observationRegistry, searchObservationConvention);
 		}
 
 	}
 
+	@Override
+	public VectorStoreObservationContext.Builder createObservationContextBuilder(String operationName) {
+
+		return VectorStoreObservationContext.builder("pg_vector", operationName)
+			.withDimensions(this.embeddingDimensions())
+			.withCollectionName(this.vectorTableName)
+			.withNamespace(this.schemaName)
+			.withSimilarityMetric(this.getDistanceType().name())
+			.withIndexName(this.createIndexMethod.name())
+			.withModel(this.embeddingModel.getClass().getSimpleName());
```

Comment:
This attribute should be the model name used for computing embedding (e.g. `text-embedding-3-small`). It's optional, so if we cannot get the value for now, I would leave this field empty. Also, since we have already instrumented the `EmbeddingModel`, we already have the model name in the trace as part of the embedding span, so we might even argue whether we want/need this here. It might be needed for cases where it's the vector database itself providing the embedding model, but as far as I know it's not something supported by the Spring AI APIs yet.

Reply:
I left this open, until the Options PR is merged. 
But even now it the `EmbeddingModel` does not expose `getDefaultOptions` method (similar to the ChatOptions) to show, at least, the default method name. 
So we should either add the get options to the EmbeddingModel or drop this attribute all together.

Reply:
I've decide to remove the Model attribute for now. We can consider adding it later if needed.